.github/workflows/test.yaml

```diff
@@ -0,0 +1,47 @@
+name: Testing CI pipeline
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build the docker image
+        run: docker-compose build
+
+      - name: up postgresql container runs
+        run: docker-compose up -d
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@ccb4328a959376b642e027874838f60f8e596de3
+      - name: Run chmod to make gradlew executable
+        run:  chmod +x gradlew
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: build
+      - name: Execute tests
+        uses: gradle/gradle-build-action@749f47bda3e44aa060e82d7b3ef7e40d953bd629
+        with:
+          arguments: test
```

Comment:
It would be nice to get like results from this step, like a test report or something, but that's not part of this issue imo but something for a later sprint. (I don't know if that is even possible)